connectivity/lwipstack/source/LWIPStack.cpp

```diff
@@ -664,6 +664,12 @@ nsapi_error_t LWIP::setsockopt(nsapi_socket_t handle, int level, int optname, co
             return err_remap(igmp_err);
         }
 
+        case NSAPI_IPTOS:
+            if (optlen != sizeof(int)) {
```

Comment:
Why do we need an int ? What are the expected values ?

Reply:
Hi @pan- 

The IPTOS ( is a field an IP header field ) of  8 bits ( with 6 bits (DSCP) + 2 bits (ECN)). The typical values of these used Priority differentiation following the RFC 791 https://tools.ietf.org/html/rfc791 


      Bits 0-2:  Precedence.
      Bit    3:  0 = Normal Delay,      1 = Low Delay.
      Bits   4:  0 = Normal Throughput, 1 = High Throughput.
      Bits   5:  0 = Normal Relibility, 1 = High Relibility.
      Bit  6-7:  Reserved for Future Use.

         0         1        2     3      4       5         6         7
      +----+-----+------+-----+-----+-----+-----+------+
      |                             |        |          |         |        |            |
      |   PRECEDENCE     |  D   |    T    |   R    |  0    |     0     |
      |                             |        |          |         |        |            |
      +-----+-----+-----+-----+-----+-----+-----+-------+

This is used to priority on Traffic class in transmit direction the following are the values used

uint32_t tos_priority[4] = { 0x00, 0x40, 0xA0, 0xE0 }; // Best Effort, Background, Video, Voice


Regarding add it as nsapi_types.h , please can give more information, I know that `NSAPI_ADD_MEMBERSHIP/NSAPI_DROP_MEMBERSHIP` is only used for muliticast traffic and how I should add, the following is used for unicast traffic.

Thanks
Balaji.

Reply:
@balajicyp that doesn't really answer the question of why do we need an int as a single byte is required to code all the values. 

For the options what I'd like to see is a list of these flags in `NSAPI`. Something similar to what is presented in LwIP: https://github.com/ARMmbed/mbed-os/blob/ad40b1b2672d59b058f7c104fdc612d0dddf3ac3/connectivity/lwipstack/lwip/src/include/lwip/sockets.h#L347-L389

The goal is to not let the user guess what can be used or hardcode these values. Other stacks may it should be clear what values are expected.

Reply:
@pan- 

I have added the NSAPI type defines for `Type of Service` similar to mbed-os/connectivity/lwipstack/lwip/src/include/lwip/sockets.h (line 347 to 389). 

I have fixed optlen check to u8_t

Please review them

Thanks
Balaji.